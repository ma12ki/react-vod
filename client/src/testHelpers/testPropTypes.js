import React from 'react';

// this has some limitations:
// https://stackoverflow.com/questions/41916992/why-does-my-jest-test-of-react-proptypes-break-when-using-multiple-unacceptable/41916993#41916993
export const testPropTypes = (component, propName, acceptableValues, unacceptableValues, otherProps) => {
    console.error = jest.fn();
    const _test = (testValues, expectError) => {
        for (let propValue of testValues) {
            console.error.mockClear();
            React.createElement(component, {...otherProps, [propName]: propValue});
            try {
                expect(console.error).toHaveBeenCalledTimes(expectError ? 1 : 0);
            } catch (err) {
                console.log(`${propName} expected to ${expectError ? '' : 'NOT'} fail for value ${propValue}`);
                throw err;
            }
        }
    };
    _test(acceptableValues || [], false);
    _test(unacceptableValues || [], true);
};
