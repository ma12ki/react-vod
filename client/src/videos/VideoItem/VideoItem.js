import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Link from 'redux-first-router-link';
import moment from 'moment';
import FA from 'react-fontawesome';

import { routesKeys } from '../routes';
import { FileDuration, FileSize, HighlightOccurences } from '../../shared';
import { getSearch } from '../selectors';
import { utils } from '../../shared/cssModules';

export const VideoItem = ({ video, searchTerm }) => {
    const { id, name, title, path, size = 0, duration = 0, dateCreated = new Date(0) } = video; 

    return (
        <tr>
            <td title={title + '; ' + path} className={utils.alignLeft}>
                <HighlightOccurences term={searchTerm}>{name}</HighlightOccurences>
            </td>
            <td className={utils.center}>{moment(dateCreated).format('DD.MM.YYYY')}</td>
            <td className={utils.alignRight}><FileDuration duration={duration} /></td>
            <td className={utils.alignRight}><FileSize size={size} /></td>
            <td className={utils.center}>
                <Link to={{ type: routesKeys.video, payload: { id } }}>
                    <FA name='play' />
                </Link>
            </td>
        </tr>
    );
};

VideoItem.propTypes = {
    video: PropTypes.shape({
        id: PropTypes.string,
        name: PropTypes.string,
        title: PropTypes.string,
        path: PropTypes.string,
        size: PropTypes.number,
        duration: PropTypes.number,
        dateCreated: PropTypes.any,
    }).isRequired,
    searchTerm: PropTypes.string,
};

const mapStateToProps = (state) => ({
    searchTerm: getSearch(state),
});

export default connect(mapStateToProps)(VideoItem);
