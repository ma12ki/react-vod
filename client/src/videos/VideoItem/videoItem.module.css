.row {
    cursor: pointer;
    background-color: transparent;
    transition: background-color 0.15s ease-in-out;
}

.row:global(:hover) {
    background-color: rgba(2, 117, 216, 0.3);
}
