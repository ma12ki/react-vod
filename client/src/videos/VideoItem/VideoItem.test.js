import React from 'react';
import { shallow } from 'enzyme';

import { testPropTypes } from '../../testHelpers';
import { VideoItem } from './VideoItem';

const makeWrapper = ({ video = {}, searchTerm = '' } = {}) => shallow(<VideoItem video={video} searchTerm={searchTerm} />);

describe('VideoItem', () => {

    it('matches the snapshot', () => {
        const wrapper = makeWrapper();

        expect(wrapper).toMatchSnapshot();
    });

    it('accepts "video" object prop', () => {
        testPropTypes(
            VideoItem,
            'video',
            [{}],
            [123, '123', null, undefined],
            { searchTerm: '' }
        );
    });

    it('accepts "searchTerm" string prop', () => {
        testPropTypes(
            VideoItem,
            'searchTerm',
            ['ad sd fsd f sdf sdf sdfsd', 'FFFFF!@#$%^&*()_', '', null, undefined],
            [123, <i></i>],
            { video: {} }
        );
    });

});