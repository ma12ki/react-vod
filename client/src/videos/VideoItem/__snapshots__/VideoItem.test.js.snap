// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr>
    <td
        className="alignLeft"
        title="undefined; undefined"
    >
        <Unknown
            term=""
        />
    </td>
    <td
        className="center"
    >
        01.01.1970
    </td>
    <td
        className="alignRight"
    >
        <Unknown
            duration={0}
        />
    </td>
    <td
        className="alignRight"
    >
        <Unknown
            size={0}
        />
    </td>
    <td
        className="center"
    >
        <Connect(Link)
            to={
                Object {
                    "payload": Object {
                      "id": undefined,
                    },
                    "type": "vod/router/videos/VIDEO",
                  }
            }
        >
            <FontAwesome
                name="play"
            />
        </Connect(Link)>
    </td>
</tr>,
  "nodes": Array [
    <tr>
      <td
            className="alignLeft"
            title="undefined; undefined"
      >
            <Unknown
                  term=""
            />
      </td>
      <td
            className="center"
      >
            01.01.1970
      </td>
      <td
            className="alignRight"
      >
            <Unknown
                  duration={0}
            />
      </td>
      <td
            className="alignRight"
      >
            <Unknown
                  size={0}
            />
      </td>
      <td
            className="center"
      >
            <Connect(Link)
                  to={
                        Object {
                              "payload": Object {
                                "id": undefined,
                              },
                              "type": "vod/router/videos/VIDEO",
                            }
                  }
            >
                  <FontAwesome
                        name="play"
                  />
            </Connect(Link)>
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        searchTerm=""
        video={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "searchTerm": "",
          "video": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr>
          <td
                    className="alignLeft"
                    title="undefined; undefined"
          >
                    <Unknown
                              term=""
                    />
          </td>
          <td
                    className="center"
          >
                    01.01.1970
          </td>
          <td
                    className="alignRight"
          >
                    <Unknown
                              duration={0}
                    />
          </td>
          <td
                    className="alignRight"
          >
                    <Unknown
                              size={0}
                    />
          </td>
          <td
                    className="center"
          >
                    <Connect(Link)
                              to={
                                        Object {
                                                  "payload": Object {
                                                    "id": undefined,
                                                  },
                                                  "type": "vod/router/videos/VIDEO",
                                                }
                              }
                    >
                              <FontAwesome
                                        name="play"
                              />
                    </Connect(Link)>
          </td>
</tr>,
        "_debugID": 2,
        "_renderedOutput": <tr>
          <td
                    className="alignLeft"
                    title="undefined; undefined"
          >
                    <Unknown
                              term=""
                    />
          </td>
          <td
                    className="center"
          >
                    01.01.1970
          </td>
          <td
                    className="alignRight"
          >
                    <Unknown
                              duration={0}
                    />
          </td>
          <td
                    className="alignRight"
          >
                    <Unknown
                              size={0}
                    />
          </td>
          <td
                    className="center"
          >
                    <Connect(Link)
                              to={
                                        Object {
                                                  "payload": Object {
                                                    "id": undefined,
                                                  },
                                                  "type": "vod/router/videos/VIDEO",
                                                }
                              }
                    >
                              <FontAwesome
                                        name="play"
                              />
                    </Connect(Link)>
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    searchTerm=""
    video={Object {}}
/>,
}
`;
